SELinux libs and tools: 2.4 -> 2.7

xeji opened the pull request:
###### Motivation for this change

Our SELinux 2.4 is >3 yrs old and produces some build fails with gcc7.
[WIP] since this update involves multiple packages which will be added here incrementally.

/cc @phreedom @andir

xeji committed:
libsepol: 2.4 -> 2.7

Mic92 commented:
Has anyone a working selinux on NixOS actually? I did not even manage to enable it.

xeji commented:
I want to  give it a try, that's why I am doing this first.

xeji committed:
libselinux: 2.4 -> 2.7

xeji committed:
libsemanage: 2.4 -> 2.7

xeji committed:
policycoreutils: 2.4 -> 2.7

xeji committed:
checkpolicy: 2.4 -> 2.7

Mic92 commented:
I would be glad to see this. I currently have to boot fedora to test selinux support in https://github.com/Mic92/cntr

Mic92 commented:
I would not be surprise that I was not able to enable it because of the age of selinux in comparison to the kernel.

Mic92 commented:
@GrahamcOfBorg build libsepool libselinux libsemanage policycoreutils checkpolicy

GrahamcOfBorg commented:
No attempt on x86_64-darwin

The following builds were skipped because they don't evaluate on x86_64-darwin: libsepool, libselinux, libsemanage, policycoreutils, checkpolicy

No log is available.

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.36978&attempt_id=78410985-a7b8-4cdb-9d83-f5bb72cf595e)

Attempted: libselinux, libsemanage, policycoreutils, checkpolicy

The following builds were skipped because they don't evaluate on x86_64-linux: libsepool

<details><summary>Partial log (click to expand)</summary><p>

```
strip is /nix/store/b0zlxla7dmy1iwc3g459rjznx59797xy-binutils-2.28.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/276dvgnylyflqnv8f8gs9j4r4yvf8s5i-policycoreutils-2.7/libexec  /nix/store/276dvgnylyflqnv8f8gs9j4r4yvf8s5i-policycoreutils-2.7/bin  /nix/store/276dvgnylyflqnv8f8gs9j4r4yvf8s5i-policycoreutils-2.7/sbin
patching script interpreter paths in /nix/store/276dvgnylyflqnv8f8gs9j4r4yvf8s5i-policycoreutils-2.7
/nix/store/276dvgnylyflqnv8f8gs9j4r4yvf8s5i-policycoreutils-2.7/sbin/fixfiles: interpreter directive changed from "/bin/bash" to "/nix/store/zqh3l3lyw32q1ayb15bnvg9f24j5v2p0-bash-4.4-p12/bin/bash"
checking for references to /build in /nix/store/276dvgnylyflqnv8f8gs9j4r4yvf8s5i-policycoreutils-2.7...
moving /nix/store/276dvgnylyflqnv8f8gs9j4r4yvf8s5i-policycoreutils-2.7/sbin/* to /nix/store/276dvgnylyflqnv8f8gs9j4r4yvf8s5i-policycoreutils-2.7/bin
/nix/store/zw1m95k8rhhzn13ipz3p5b56gb4qr93w-libselinux-2.7
/nix/store/xyi7sq79bkpx6wwqqbgjp3l3rasd3c7q-libsemanage-2.7
/nix/store/276dvgnylyflqnv8f8gs9j4r4yvf8s5i-policycoreutils-2.7
/nix/store/xzifskgagi70z9cw77z9cib1mfd59vv0-checkpolicy-2.7
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.36978&attempt_id=3e5cb003-822e-4b57-82a7-2cee60bcbfae)

Attempted: libselinux, libsemanage, policycoreutils, checkpolicy

The following builds were skipped because they don't evaluate on aarch64-linux: libsepool

<details><summary>Partial log (click to expand)</summary><p>

```
strip is /nix/store/lvx1acn1ig1j2km8jds5x3ggh3f2wa8v-binutils-2.28.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/hq0m5kfq923299n3k6sh8mjm63b5aq56-policycoreutils-2.7/libexec  /nix/store/hq0m5kfq923299n3k6sh8mjm63b5aq56-policycoreutils-2.7/bin  /nix/store/hq0m5kfq923299n3k6sh8mjm63b5aq56-policycoreutils-2.7/sbin
patching script interpreter paths in /nix/store/hq0m5kfq923299n3k6sh8mjm63b5aq56-policycoreutils-2.7
/nix/store/hq0m5kfq923299n3k6sh8mjm63b5aq56-policycoreutils-2.7/sbin/fixfiles: interpreter directive changed from "/bin/bash" to "/nix/store/lnws52qanbi5vz8qg24v4ym57nznfndg-bash-4.4-p12/bin/bash"
checking for references to /build in /nix/store/hq0m5kfq923299n3k6sh8mjm63b5aq56-policycoreutils-2.7...
moving /nix/store/hq0m5kfq923299n3k6sh8mjm63b5aq56-policycoreutils-2.7/sbin/* to /nix/store/hq0m5kfq923299n3k6sh8mjm63b5aq56-policycoreutils-2.7/bin
/nix/store/k39qmghs50z7igga6bcplx2jqy8syjfx-libselinux-2.7
/nix/store/0sr2ywbc4j62z2yfsagj6f20fn8zpd88-libsemanage-2.7
/nix/store/hq0m5kfq923299n3k6sh8mjm63b5aq56-policycoreutils-2.7
/nix/store/mh6gcyvsamz0icnbnkq3144mlsip4v31-checkpolicy-2.7
```
</p></details>

xeji committed:
selinux-sandbox: init at 2.7

xeji commented:
@GrahamcOfBorg build libsepol libselinux libsemanage policycoreutils checkpolicy

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.36978&attempt_id=b95f4427-1380-4cbc-aeba-a665a8b5555c)

Attempted: libsepol, libselinux, libsemanage, policycoreutils, checkpolicy

<details><summary>Partial log (click to expand)</summary><p>

```
stripping (with command strip and flags -S) in /nix/store/n6lb7r5yxccgdq9z7wvp3v8pzbj3kv8w-policycoreutils-2.7/libexec  /nix/store/n6lb7r5yxccgdq9z7wvp3v8pzbj3kv8w-policycoreutils-2.7/bin  /nix/store/n6lb7r5yxccgdq9z7wvp3v8pzbj3kv8w-policycoreutils-2.7/sbin
patching script interpreter paths in /nix/store/n6lb7r5yxccgdq9z7wvp3v8pzbj3kv8w-policycoreutils-2.7
/nix/store/n6lb7r5yxccgdq9z7wvp3v8pzbj3kv8w-policycoreutils-2.7/sbin/fixfiles: interpreter directive changed from "/bin/bash" to "/nix/store/q1g0rl8zfmz7r371fp5p42p4acmv297d-bash-4.4-p19/bin/bash"
checking for references to /build in /nix/store/n6lb7r5yxccgdq9z7wvp3v8pzbj3kv8w-policycoreutils-2.7...
moving /nix/store/n6lb7r5yxccgdq9z7wvp3v8pzbj3kv8w-policycoreutils-2.7/sbin/* to /nix/store/n6lb7r5yxccgdq9z7wvp3v8pzbj3kv8w-policycoreutils-2.7/bin
/nix/store/6ri8425ij5f1bjmhr0wx722dhml1ir78-libsepol-2.7
/nix/store/5gl66m6kgrpyy8i80fmbyqvsfxvj9phg-libselinux-2.7
/nix/store/0jn58hb0k4zc3jnd79pj20nbng0910jf-libsemanage-2.7
/nix/store/n6lb7r5yxccgdq9z7wvp3v8pzbj3kv8w-policycoreutils-2.7
/nix/store/3igiz2cgimbqhc2fzka0p8crv8qlmqnv-checkpolicy-2.7
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.36978&attempt_id=a79df0a0-2efd-4b94-a07d-8e6b527b61f3)

Attempted: libsepol, libselinux, libsemanage, policycoreutils, checkpolicy

<details><summary>Partial log (click to expand)</summary><p>

```
stripping (with command strip and flags -S) in /nix/store/7749nh3bly19q38srw4yjv43wi2fw2am-policycoreutils-2.7/libexec  /nix/store/7749nh3bly19q38srw4yjv43wi2fw2am-policycoreutils-2.7/bin  /nix/store/7749nh3bly19q38srw4yjv43wi2fw2am-policycoreutils-2.7/sbin
patching script interpreter paths in /nix/store/7749nh3bly19q38srw4yjv43wi2fw2am-policycoreutils-2.7
/nix/store/7749nh3bly19q38srw4yjv43wi2fw2am-policycoreutils-2.7/sbin/fixfiles: interpreter directive changed from "/bin/bash" to "/nix/store/3gg2p6n4kv2f0lsxd41f5iz1ivkbzyzr-bash-4.4-p19/bin/bash"
checking for references to /build in /nix/store/7749nh3bly19q38srw4yjv43wi2fw2am-policycoreutils-2.7...
moving /nix/store/7749nh3bly19q38srw4yjv43wi2fw2am-policycoreutils-2.7/sbin/* to /nix/store/7749nh3bly19q38srw4yjv43wi2fw2am-policycoreutils-2.7/bin
/nix/store/xvhmxmkx2jmis72nywgn5ha5qp20b8ba-libsepol-2.7
/nix/store/xrs3h0d8iby18mk0vr1wiiqmkrxnc9r3-libselinux-2.7
/nix/store/1prjv8gm5pydr2cpnhskq6738kccxnim-libsemanage-2.7
/nix/store/7749nh3bly19q38srw4yjv43wi2fw2am-policycoreutils-2.7
/nix/store/73ms89sm8kl67jf5fc5icrknmdby1n9r-checkpolicy-2.7
```
</p></details>

xeji committed:
setools: 2015-02-12 -> 2017-11-10

xeji commented:
@GrahamcOfBorg build setools

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.36978&attempt_id=0e57416d-95ed-49d1-a807-6b59e0b05b4d)

Attempted: setools

<details><summary>Partial log (click to expand)</summary><p>

```
RoleTransition default role ... ok
test_060_conditional (tests.policyrep.rbacrule.RoleTransitionTest)
RoleTransition conditional expression ... ok
test_100_statement (tests.policyrep.rbacrule.RoleTransitionTest)
RoleTransition statement. ... ok

----------------------------------------------------------------------
Ran 1358 tests in 4.913s

OK (skipped=17, expected failures=1)
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.36978&attempt_id=d624ccee-c24c-40c7-aeeb-74c7193bf9a0)

Attempted: setools

<details><summary>Partial log (click to expand)</summary><p>

```
Boolean query with exact match ... ok
test_002_name_regex (tests.boolquery.BoolQueryTest)
Boolean query with regex match. ... ok
test_010_default (tests.boolquery.BoolQueryTest)
Boolean query with default state match. ... ok

----------------------------------------------------------------------
Ran 1358 tests in 27.757s

OK (skipped=17, expected failures=1)
```
</p></details>

xeji committed:
libsemanage: fix python lib path

xeji committed:
libselinux: fix python lib path

xeji committed:
selinux-python: init at 2.7

tools written in python that upstream moved from policycoreutils
into a separate package

xeji commented:
@grahamcofborg build selinux-python

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.36978&attempt_id=8b4e006d-904c-4fb1-b264-3b9f88f51553)

Attempted: selinux-python

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/5hhzhs1z8dl2h09sm018ygvyg8nn7433-selinux-python-2.7/bin/sepolicy: interpreter directive changed from " /usr/bin/python -Es" to "/nix/store/b8gd0cbvkm59x8flbc53bvsvmskyig5a-python3-3.6.4/bin/python -Es"
/nix/store/5hhzhs1z8dl2h09sm018ygvyg8nn7433-selinux-python-2.7/lib/python3.6/site-packages/seobject.py: interpreter directive changed from " /usr/bin/python -Es" to "/nix/store/b8gd0cbvkm59x8flbc53bvsvmskyig5a-python3-3.6.4/bin/python -Es"
/nix/store/5hhzhs1z8dl2h09sm018ygvyg8nn7433-selinux-python-2.7/sbin/semanage: interpreter directive changed from " /usr/bin/python -Es" to "/nix/store/b8gd0cbvkm59x8flbc53bvsvmskyig5a-python3-3.6.4/bin/python -Es"
checking for references to /tmp/nix-build-selinux-python-2.7.drv-0 in /nix/store/5hhzhs1z8dl2h09sm018ygvyg8nn7433-selinux-python-2.7...
moving /nix/store/5hhzhs1z8dl2h09sm018ygvyg8nn7433-selinux-python-2.7/sbin/* to /nix/store/5hhzhs1z8dl2h09sm018ygvyg8nn7433-selinux-python-2.7/bin
wrapping `/nix/store/5hhzhs1z8dl2h09sm018ygvyg8nn7433-selinux-python-2.7/bin/sepolgen-ifgen'...
wrapping `/nix/store/5hhzhs1z8dl2h09sm018ygvyg8nn7433-selinux-python-2.7/bin/audit2allow'...
wrapping `/nix/store/5hhzhs1z8dl2h09sm018ygvyg8nn7433-selinux-python-2.7/bin/chcat'...
wrapping `/nix/store/5hhzhs1z8dl2h09sm018ygvyg8nn7433-selinux-python-2.7/bin/semanage'...
wrapping `/nix/store/5hhzhs1z8dl2h09sm018ygvyg8nn7433-selinux-python-2.7/bin/sepolicy'...
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.36978&attempt_id=97b0dd60-ac3a-4a84-ac2b-a89531039afc)

Attempted: selinux-python

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/ajh6zd3l1w5vahl3hqhcbckakrrmb3rq-selinux-python-2.7/bin/sepolicy: interpreter directive changed from " /usr/bin/python -Es" to "/nix/store/143hazf8s5236kvprxlrgw9fdgzkrg0v-python3-3.6.4/bin/python -Es"
/nix/store/ajh6zd3l1w5vahl3hqhcbckakrrmb3rq-selinux-python-2.7/lib/python3.6/site-packages/seobject.py: interpreter directive changed from " /usr/bin/python -Es" to "/nix/store/143hazf8s5236kvprxlrgw9fdgzkrg0v-python3-3.6.4/bin/python -Es"
checking for references to /build in /nix/store/ajh6zd3l1w5vahl3hqhcbckakrrmb3rq-selinux-python-2.7...
moving /nix/store/ajh6zd3l1w5vahl3hqhcbckakrrmb3rq-selinux-python-2.7/sbin/* to /nix/store/ajh6zd3l1w5vahl3hqhcbckakrrmb3rq-selinux-python-2.7/bin
wrapping `/nix/store/ajh6zd3l1w5vahl3hqhcbckakrrmb3rq-selinux-python-2.7/bin/semanage'...
wrapping `/nix/store/ajh6zd3l1w5vahl3hqhcbckakrrmb3rq-selinux-python-2.7/bin/sepolicy'...
wrapping `/nix/store/ajh6zd3l1w5vahl3hqhcbckakrrmb3rq-selinux-python-2.7/bin/audit2allow'...
wrapping `/nix/store/ajh6zd3l1w5vahl3hqhcbckakrrmb3rq-selinux-python-2.7/bin/sepolgen-ifgen'...
wrapping `/nix/store/ajh6zd3l1w5vahl3hqhcbckakrrmb3rq-selinux-python-2.7/bin/chcat'...
/nix/store/ajh6zd3l1w5vahl3hqhcbckakrrmb3rq-selinux-python-2.7
```
</p></details>

Mic92 commented on pkgs/os-specific/linux/setools/default.nix:
```diff
@@ -1,35 +1,42 @@
-  nativeBuildInputs = [ autoreconfHook pkgconfig python swig2 bison flex ];
-  buildInputs = [ tcl libxml2 sqlite bzip2 ];
+  preBuild = ''
+    export SEPOL="${libsepol}/lib/libsepol.a"
```
Is `libsepol` supposed to be statically linked?

xeji commented:
Yes, upstream code explicitly links libsepol.a  statically. Surprised me too but I see no reason to mess with that. It's not that big anyway.

xeji commented:
@GrahamcOfBorg build selinux-sandbox

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.36978&attempt_id=03f7f0df-d71b-468a-9000-8dcb2a8d574d)

Attempted: selinux-sandbox

<details><summary>Partial log (click to expand)</summary><p>

```
strip is /nix/store/fzcs0fn6bb04m82frhlb78nc03ny3w55-binutils-2.28.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/qnqjfzw4fyr8pcspv3rdnfcplyda9dcr-selinux-sandbox-2.7/bin  /nix/store/qnqjfzw4fyr8pcspv3rdnfcplyda9dcr-selinux-sandbox-2.7/sbin
patching script interpreter paths in /nix/store/qnqjfzw4fyr8pcspv3rdnfcplyda9dcr-selinux-sandbox-2.7
/nix/store/qnqjfzw4fyr8pcspv3rdnfcplyda9dcr-selinux-sandbox-2.7/share/sandbox/start: interpreter directive changed from " /usr/bin/python -Es" to "/nix/store/b8gd0cbvkm59x8flbc53bvsvmskyig5a-python3-3.6.4/bin/python -Es"
/nix/store/qnqjfzw4fyr8pcspv3rdnfcplyda9dcr-selinux-sandbox-2.7/share/sandbox/sandboxX.sh: interpreter directive changed from "/bin/bash" to "/nix/store/q1g0rl8zfmz7r371fp5p42p4acmv297d-bash-4.4-p19/bin/bash"
/nix/store/qnqjfzw4fyr8pcspv3rdnfcplyda9dcr-selinux-sandbox-2.7/bin/sandbox: interpreter directive changed from " /usr/bin/python -Es" to "/nix/store/b8gd0cbvkm59x8flbc53bvsvmskyig5a-python3-3.6.4/bin/python -Es"
checking for references to /tmp/nix-build-selinux-sandbox-2.7.drv-0 in /nix/store/qnqjfzw4fyr8pcspv3rdnfcplyda9dcr-selinux-sandbox-2.7...
moving /nix/store/qnqjfzw4fyr8pcspv3rdnfcplyda9dcr-selinux-sandbox-2.7/sbin/* to /nix/store/qnqjfzw4fyr8pcspv3rdnfcplyda9dcr-selinux-sandbox-2.7/bin
wrapping `/nix/store/qnqjfzw4fyr8pcspv3rdnfcplyda9dcr-selinux-sandbox-2.7/bin/sandbox'...
/nix/store/qnqjfzw4fyr8pcspv3rdnfcplyda9dcr-selinux-sandbox-2.7
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.36978&attempt_id=8b358eb1-eec1-4535-8d81-b5131f797e3e)

Attempted: selinux-sandbox

<details><summary>Partial log (click to expand)</summary><p>

```
strip is /nix/store/3zq400fri5dv7d30lpxlqm2v9y1iis6j-binutils-2.28.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/dy8vlsl18x439sbydss4bw0718gfdc7n-selinux-sandbox-2.7/bin  /nix/store/dy8vlsl18x439sbydss4bw0718gfdc7n-selinux-sandbox-2.7/sbin
patching script interpreter paths in /nix/store/dy8vlsl18x439sbydss4bw0718gfdc7n-selinux-sandbox-2.7
/nix/store/dy8vlsl18x439sbydss4bw0718gfdc7n-selinux-sandbox-2.7/share/sandbox/start: interpreter directive changed from " /usr/bin/python -Es" to "/nix/store/143hazf8s5236kvprxlrgw9fdgzkrg0v-python3-3.6.4/bin/python -Es"
/nix/store/dy8vlsl18x439sbydss4bw0718gfdc7n-selinux-sandbox-2.7/share/sandbox/sandboxX.sh: interpreter directive changed from "/bin/bash" to "/nix/store/3gg2p6n4kv2f0lsxd41f5iz1ivkbzyzr-bash-4.4-p19/bin/bash"
/nix/store/dy8vlsl18x439sbydss4bw0718gfdc7n-selinux-sandbox-2.7/bin/sandbox: interpreter directive changed from " /usr/bin/python -Es" to "/nix/store/143hazf8s5236kvprxlrgw9fdgzkrg0v-python3-3.6.4/bin/python -Es"
checking for references to /build in /nix/store/dy8vlsl18x439sbydss4bw0718gfdc7n-selinux-sandbox-2.7...
moving /nix/store/dy8vlsl18x439sbydss4bw0718gfdc7n-selinux-sandbox-2.7/sbin/* to /nix/store/dy8vlsl18x439sbydss4bw0718gfdc7n-selinux-sandbox-2.7/bin
wrapping `/nix/store/dy8vlsl18x439sbydss4bw0718gfdc7n-selinux-sandbox-2.7/bin/sandbox'...
/nix/store/dy8vlsl18x439sbydss4bw0718gfdc7n-selinux-sandbox-2.7
```
</p></details>

xeji commented:
@grahamcofborg build libsepol libselinux libsemanage policycoreutils checkpolicy

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.36978&attempt_id=16f62b16-6eeb-4a73-b4e4-a690a7b780e6)

Attempted: libsepol, libselinux, libsemanage, policycoreutils, checkpolicy

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/6ri8425ij5f1bjmhr0wx722dhml1ir78-libsepol-2.7
/nix/store/5gl66m6kgrpyy8i80fmbyqvsfxvj9phg-libselinux-2.7
/nix/store/0jn58hb0k4zc3jnd79pj20nbng0910jf-libsemanage-2.7
/nix/store/n6lb7r5yxccgdq9z7wvp3v8pzbj3kv8w-policycoreutils-2.7
/nix/store/3igiz2cgimbqhc2fzka0p8crv8qlmqnv-checkpolicy-2.7
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.36978&attempt_id=3cbdd3ae-71a9-4131-aaf1-45b625a18c3b)

Attempted: libsepol, libselinux, libsemanage, policycoreutils, checkpolicy

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/xvhmxmkx2jmis72nywgn5ha5qp20b8ba-libsepol-2.7
/nix/store/xrs3h0d8iby18mk0vr1wiiqmkrxnc9r3-libselinux-2.7
/nix/store/1prjv8gm5pydr2cpnhskq6738kccxnim-libsemanage-2.7
/nix/store/7749nh3bly19q38srw4yjv43wi2fw2am-policycoreutils-2.7
/nix/store/73ms89sm8kl67jf5fc5icrknmdby1n9r-checkpolicy-2.7
```
</p></details>

xeji commented:
These look good now. Not yet tested on an SELinux-enabled system though (I somehow doubt our current version ever was).

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.36978&attempt_id=d817af72-df44-457a-a0b8-20f58188b0e5)

Attempted: checkpolicy, libselinux, libsemanage, libsepol, policycoreutils, selinux-python, selinux-sandbox, setools

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/3igiz2cgimbqhc2fzka0p8crv8qlmqnv-checkpolicy-2.7
/nix/store/5gl66m6kgrpyy8i80fmbyqvsfxvj9phg-libselinux-2.7
/nix/store/0jn58hb0k4zc3jnd79pj20nbng0910jf-libsemanage-2.7
/nix/store/6ri8425ij5f1bjmhr0wx722dhml1ir78-libsepol-2.7
/nix/store/n6lb7r5yxccgdq9z7wvp3v8pzbj3kv8w-policycoreutils-2.7
/nix/store/5hhzhs1z8dl2h09sm018ygvyg8nn7433-selinux-python-2.7
/nix/store/qnqjfzw4fyr8pcspv3rdnfcplyda9dcr-selinux-sandbox-2.7
/nix/store/74vmcgkg3vk8siz7f58qp5bz9lyjwibp-setools-2017-11-10
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.36978&attempt_id=f8dd120e-0d94-468b-9b0a-55c670746d05)

Attempted: checkpolicy, libselinux, libsemanage, libsepol, policycoreutils, selinux-python, selinux-sandbox, setools

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/73ms89sm8kl67jf5fc5icrknmdby1n9r-checkpolicy-2.7
/nix/store/xrs3h0d8iby18mk0vr1wiiqmkrxnc9r3-libselinux-2.7
/nix/store/1prjv8gm5pydr2cpnhskq6738kccxnim-libsemanage-2.7
/nix/store/xvhmxmkx2jmis72nywgn5ha5qp20b8ba-libsepol-2.7
/nix/store/7749nh3bly19q38srw4yjv43wi2fw2am-policycoreutils-2.7
/nix/store/ajh6zd3l1w5vahl3hqhcbckakrrmb3rq-selinux-python-2.7
/nix/store/dy8vlsl18x439sbydss4bw0718gfdc7n-selinux-sandbox-2.7
/nix/store/61mk7sw3zbdd83sigax1qbdnik0zrbax-setools-2017-11-10
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.36978&attempt_id=702603c9-93c7-433f-bf30-4b18b0886089)

Attempted: checkpolicy, libselinux, libsemanage, libsepol, policycoreutils, selinux-python, selinux-sandbox, setools

<details><summary>Partial log (click to expand)</summary><p>

```
moving /nix/store/1wh7qmmwzgmqs0vsg8yzbnbd36gbxrag-selinux-sandbox-2.7/sbin/* to /nix/store/1wh7qmmwzgmqs0vsg8yzbnbd36gbxrag-selinux-sandbox-2.7/bin
wrapping `/nix/store/1wh7qmmwzgmqs0vsg8yzbnbd36gbxrag-selinux-sandbox-2.7/bin/sandbox'...
/nix/store/3igiz2cgimbqhc2fzka0p8crv8qlmqnv-checkpolicy-2.7
/nix/store/5gl66m6kgrpyy8i80fmbyqvsfxvj9phg-libselinux-2.7
/nix/store/0jn58hb0k4zc3jnd79pj20nbng0910jf-libsemanage-2.7
/nix/store/6ri8425ij5f1bjmhr0wx722dhml1ir78-libsepol-2.7
/nix/store/n6lb7r5yxccgdq9z7wvp3v8pzbj3kv8w-policycoreutils-2.7
/nix/store/5hhzhs1z8dl2h09sm018ygvyg8nn7433-selinux-python-2.7
/nix/store/1wh7qmmwzgmqs0vsg8yzbnbd36gbxrag-selinux-sandbox-2.7
/nix/store/74vmcgkg3vk8siz7f58qp5bz9lyjwibp-setools-2017-11-10
```
</p></details>

xeji commented:
rebased to solve merge conflict

dtzWill commented:
:+1:

Mic92 commented:
@xeji did you manage to get selinux running on nixos and if what flags where required?

xeji commented:
@Mic92 not yet - it's on my list for the next few days. Testing on fedora would be good anyway to confirm the tools work independent of nixos setup.

xeji commented:
@Mic92 some quick first tests show you need ` boot.kernelParams = [ "security=selinux" "selinux=1" ];` to boot nixos with selinux. Then the logs look promising: 
 
```
# dmesg | grep -i selinux
[    0.001588] SELinux:  Initializing.
[    0.001592] SELinux:  Starting in permissive mode
[    0.382732] SELinux:  Registering netfilter hooks
[    1.574033] systemd[1]: systemd 237 running in system mode. (+PAM +AUDIT -SELINUX +IMA +APPARMOR +SMACK -SYSVINIT +UTMP -LIBCRYPTSETUP +GCRYPT -GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID -ELFUTILS +KMOD +IDN2 -IDN -PCRE2 default-hierarchy=hybrid)
```
However, after boot selinux still shows as disabled: 
```
# sestatus
SELinux status:                 disabled
```
I suspect the `-SELINUX` in systemd build config prevents it from working, so we would probably have to override and rebuild systemd. Don't know how to do that (it's a separate NixOS repo ?). 
Anyway, this situation looks already better than selinux 2.4 which caused systemd-related kernel backtraces...

dtzWill commented:
> I suspect the -SELINUX in systemd build config prevents it from working, so we would probably have to override and rebuild systemd. Don't know how to do that (it's a separate NixOS repo ?).
> Anyway, this situation looks already better than selinux 2.4 which caused systemd-related kernel backtraces...

I believe that means it wasn't available at compile-time (or that we forced it off with `-Dselinux=false`, which we don't)-- which seems like a good starting point.  No clue what else might need changing, if anything, though.

Also, yay!

Mic92 commented:
@xeji looking at systemd is a good hint. It is possible to enable selinux in selinux without recompiling everything by loading a custom systemd with `systemd.package =` in `configuration.nix`. I will skim over the code to see if I can find something interesting.

xeji commented:
Looking at the systemd expression, we don't disable selinux. But we don't list `libselinux` (that's what I guess it would be) as build input either. So if we're lucky, just adding that may hopefully configure systemd with `+SELINUX`. I'll give that a try when I have time for a lengthy build...

Mic92 commented:
`selinux_init_load_policy` seems to enable selinux globally: 
https://github.com/systemd/systemd/blob/master/src/core/selinux-setup.c#L81

xeji commented:
https://github.com/systemd/systemd/blob/ffb3c2bd705409ca5bbbb9ccef4c59349ea787cf/meson.build#L850 
if libselinux is present, systemd builds with it

xeji committed:
systemd: add withSelinux option

false by default, so no rebuild

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.36978&attempt_id=50493e2b-0bf3-4317-ad1a-8dd2a051ac13)

Attempted: checkpolicy, libselinux, libsemanage, libsepol, policycoreutils, selinux-python, selinux-sandbox, setools, systemd

<details><summary>Partial log (click to expand)</summary><p>

```
wrapping `/nix/store/1wh7qmmwzgmqs0vsg8yzbnbd36gbxrag-selinux-sandbox-2.7/bin/sandbox'...
/nix/store/3igiz2cgimbqhc2fzka0p8crv8qlmqnv-checkpolicy-2.7
/nix/store/5gl66m6kgrpyy8i80fmbyqvsfxvj9phg-libselinux-2.7
/nix/store/0jn58hb0k4zc3jnd79pj20nbng0910jf-libsemanage-2.7
/nix/store/6ri8425ij5f1bjmhr0wx722dhml1ir78-libsepol-2.7
/nix/store/n6lb7r5yxccgdq9z7wvp3v8pzbj3kv8w-policycoreutils-2.7
/nix/store/5hhzhs1z8dl2h09sm018ygvyg8nn7433-selinux-python-2.7
/nix/store/1wh7qmmwzgmqs0vsg8yzbnbd36gbxrag-selinux-sandbox-2.7
/nix/store/74vmcgkg3vk8siz7f58qp5bz9lyjwibp-setools-2017-11-10
/nix/store/aw6jabbci1k81j6hvyskkhb3va45wws2-systemd-237
```
</p></details>

xeji commented:
The last commit adds a `withSelinux` option to systemd to build it with selinux support, `false` by default so this should not trigger a rebuild. Now  nixos boots with selinux enabled using [this config](https://gist.github.com/xeji/78d4abbb493fc5a5b6f069b41ebdafbe).
For a fully functional selinux-enabled system we might also have to rebuild other packages like `coreutils` and `util-linux`  with selinux support according to the [Arch wiki](https://wiki.archlinux.org/index.php/SELinux).

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.36978&attempt_id=da046901-f3b3-4150-813e-eb9df991c5a6)

Attempted: checkpolicy, libselinux, libsemanage, libsepol, policycoreutils, selinux-python, selinux-sandbox, setools, systemd

<details><summary>Partial log (click to expand)</summary><p>

```
wrapping `/nix/store/mssq69yg4livphqx3ms1fqnyj6m8fman-selinux-sandbox-2.7/bin/sandbox'...
/nix/store/73ms89sm8kl67jf5fc5icrknmdby1n9r-checkpolicy-2.7
/nix/store/xrs3h0d8iby18mk0vr1wiiqmkrxnc9r3-libselinux-2.7
/nix/store/1prjv8gm5pydr2cpnhskq6738kccxnim-libsemanage-2.7
/nix/store/xvhmxmkx2jmis72nywgn5ha5qp20b8ba-libsepol-2.7
/nix/store/7749nh3bly19q38srw4yjv43wi2fw2am-policycoreutils-2.7
/nix/store/ajh6zd3l1w5vahl3hqhcbckakrrmb3rq-selinux-python-2.7
/nix/store/mssq69yg4livphqx3ms1fqnyj6m8fman-selinux-sandbox-2.7
/nix/store/61mk7sw3zbdd83sigax1qbdnik0zrbax-setools-2017-11-10
/nix/store/z7hj2mx7n4j6vpzqssr4bsc6anxdi58x-systemd-237
```
</p></details>

Mic92 commented:
@xeji I suppose this selinux snippet could become a module later.

pbogdan commented:
Just a heads up - it seems `libsepol` doesn't build on i686-linux (https://hydra.nixos.org/build/72006419) which is holding up the build of the minimal iso for that platform (part of the unstable channel tested job). No idea how to fix it though :-(.

xeji commented:
@pbogdan thanks. The error looks like it can be fixed by ignoring a warning, I'll give that a try later, cannot hurt. But I have no idea how to test a i686 build. Is there a way to cross-build this on a x86_64 machine? Never tried...

pbogdan commented:
Yeah, at least for i686-linux you can build it with `nix-build -A libsepol --argstr system i686-linux`.

xeji commented:
@Mic92 a selinux module would be nice, let's keep that in mind. Probably quite a bit of work since we need some sort of policy builder/patcher tool to create policies that fit a nixos system.

Mic92 commented:
@xeji I think supporting containers would be already useful. docker/lxc/systemd-nspawn already apply profiles.

Infinisil commented on pkgs/os-specific/linux/selinux-sandbox/default.nix:
```diff
@@ -0,0 +1,61 @@
+    substituteInPlace Makefile --replace "-m 4755" "-m 755"
+    substituteInPlace sandboxX.sh \
+      --replace "#!/bin/sh" "#!${bash}/bin/sh" \
+      --replace "/usr/share/sandbox/start" "${placeholder "out"}/share/sandbox/start" \
```
This doesn't work for Nix 1.11 users, please correct this as seen in https://github.com/NixOS/nixpkgs/pull/37860

xeji commented:
@Infinisil why does it matter? That placeholder is in a new package so this shouldn't break anything nix 1.11 users had before, or does it?

On March 28, 2018 5:55:53 PM GMT+02:00, Silvan Mosberger <notifications@github.com> wrote:
>Infinisil commented on this pull request.
>
>
>
>> +
>+  src = fetchurl {
>+    url =
>"https://raw.githubusercontent.com/wiki/SELinuxProject/selinux/files/releases/${se_release}/selinux-sandbox-${version}.tar.gz";
>+    sha256 = "0hf5chm90iapb42njaps6p5460ys3ajh5446ja544vdbh01n544l";
>+  };
>+
>+  nativeBuildInputs = [ wrapPython ];
>+  buildInputs = [ bash coreutils libcap_ng policycoreutils python3
>xorgserver openbox xmodmap dbus ];
>+  propagatedBuildInputs = [ pygobject3 selinux-python ];
>+
>+  postPatch = ''
>+    # Fix setuid install
>+    substituteInPlace Makefile --replace "-m 4755" "-m 755"
>+    substituteInPlace sandboxX.sh \
>+      --replace "#!/bin/sh" "#!${bash}/bin/sh" \
>+      --replace "/usr/share/sandbox/start" "${placeholder
>"out"}/share/sandbox/start" \
>
>This doesn't work for Nix 1.11 users, please correct this as seen in
>https://github.com/NixOS/nixpkgs/pull/37860
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/NixOS/nixpkgs/pull/36978#pullrequestreview-107740602

Infinisil commented:
The minimum version required is still 1.11, not 2.0: https://github.com/NixOS/nixpkgs/blob/master/lib/minver.nix

jtojnar commented:
Fixed in 025881c2364

xeji commented:
Thanks @jtojnar !
